cmd/loadTests.go

```diff
@@ -230,8 +238,13 @@ func setup(cmd *cobra.Command, args []string) {
 		for userIndex := range chUsers {
 			startTime := time.Now()
 			username := fmt.Sprintf("%s-%04d", usernamePrefix, userIndex)
-			usernamespace := fmt.Sprintf("%s-tenant", username)
-			_, errors := framework.AsKubeAdmin.CommonController.CreateRegistryAuthSecret(
+			framework, err := framework.NewFramework(username)
```

Comment:
With this approach you actually create 3 instances of `Framework` (in Application phase, in Resources phase and in Pipelines phase) which means you would have numberOfUsers*3 instances created (including all of the stuff that comes with it). 

What about instead of creating an instance of `Framework` every time let's keep a map of frameworks per username - that way if a `Framework` (and related client) was already created for the user, let's just use it, that would save 66% of the `framework.NewFramework` calls

Reply:
> Does the libraries under the hood handle refreshing user access token if it expires? Access token expires in 900 seconds (at least as per my tests) which might happen when waiting for slow pipelines.

it handles the refreshing of token

Reply:
cool will change